git/diff.py

```diff
@@ -49,6 +49,7 @@ def diff(self, other=Index, paths=None, create_patch=False, **kwargs):
             If None, we will be compared to the working tree.
             If Treeish, it will be compared against the respective tree
             If Index ( type ), it will be compared against the index.
+            If the string 'root', it will compare the empty tree against this tree.
```

Comment:
That's a very good argument against it indeed. Rather than using a special string here, I've now opted for a special object, a little bit like the Index class. New usage:

    from git import NULL_TREE
    commit.diff(NULL_TREE)

Fixed in 82b533f86cf86c96a16f96c815533bdda0585f48. What do you think of this API?